client/components/theme-collection/index.tsx

```diff
@@ -0,0 +1,112 @@
+import { Button } from '@wordpress/components';
+import { Icon, chevronLeft, chevronRight } from '@wordpress/icons';
+import { ReactElement, useEffect, useRef } from 'react';
+import './style.scss';
+import Swiper from 'swiper';
+import { Navigation, Keyboard, Mousewheel } from 'swiper/modules';
+import { ThemeBlock } from 'calypso/components/themes-list';
+import 'swiper/css';
+import 'swiper/css/navigation';
+import 'swiper/css/keyboard';
+import 'swiper/css/mousewheel';
+
+interface ThemeCollectionProps {
+	heading: string;
+	subheading: ReactElement;
+	themes: Array< never >;
+	collectionSlug: string;
+	getScreenshotUrl: ( themeId: string ) => string;
+	siteId: string | null;
+	getButtonOptions: () => void;
+	getActionLabel: () => string;
+	isActive: () => boolean;
+	getPrice: () => string;
+	isInstalling: () => boolean;
+}
+
+export default function ThemeCollection( {
+	collectionSlug,
+	heading,
+	subheading,
+	themes,
+	getScreenshotUrl,
+	siteId,
+	getButtonOptions,
+	getActionLabel,
+	isActive,
+	getPrice,
+	isInstalling,
+}: ThemeCollectionProps ): ReactElement {
+	const swiperInstance = useRef< Swiper | null >( null );
+
+	useEffect( () => {
+		if ( themes !== null && themes.length > 0 ) {
+			const el = document.querySelector( '.swiper-container' ) as HTMLElement;
+			if ( el ) {
+				swiperInstance.current = new Swiper( el, {
+					cssMode: true,
+					mousewheel: true,
+					keyboard: {
+						enabled: true,
+						onlyInViewport: false,
+					},
+					navigation: {
+						nextEl: '.theme-collection__carousel-nav-button--next',
+						prevEl: '.theme-collection__carousel-nav-button--previous',
+					},
+					threshold: 5,
+					slideToClickedSlide: true,
+					slidesPerView: 'auto',
+					spaceBetween: 20,
+					modules: [ Navigation, Keyboard, Mousewheel ],
+				} );
+			}
+		}
+
+		return () => {
+			swiperInstance.current?.destroy();
+		};
+	}, [ themes ] );
+
+	if ( themes === null ) {
+		return <></>; // @todo Add LoadingPlaceholders instead.
```

Comment:
We should mark themes as nullable array. When themes is null, we should display a LoadingPlaceholder to avoid a flickering screen.